projects/VitalRecord/VitalRecord_util.cs

```diff
@@ -506,6 +506,10 @@ private void ValidatePartialDates(Bundle bundle)
             // If we have a value, return it
             if (part?.Value != null)
             {
+                if (part.Value is Integer)
+                {
+                    return (int?)((Integer)part.Value).Value;
```

Comment:
This is the only part I'm not 100% on. In BuildFhirDate(), the partial date values are set as `new Integer()` instead of `UnsignedInt`, and I chose this because -1 is a valid value, but that wouldn't work with an unsigned integer. However, when I change those partial date values to be `UnsignedInt` instead, it strangely doesn't fail any tests, but it seems wrong to set a value that could be -1 to be unsigned.

Reply:
Interesting that it doesn't fail. It does make sense to have it be a signed int so this looks good to me